io-engine/src/bdev/nexus/nexus_persistence.rs

```diff
@@ -71,16 +73,22 @@ pub(crate) enum PersistOp<'a> {
 }
 
 impl<'n> Nexus<'n> {
-    /// Persist information to the store.
-    pub(crate) async fn persist(&self, op: PersistOp<'_>) {
+    /// Persists nexus's information to the store.
+    pub(crate) async fn persist(&self, op: PersistOp<'_>) -> Result<(), Error> {
         if !PersistentStore::enabled() {
-            return;
+            return Ok(());
         }
 
         let mut persistent_nexus_info = self.nexus_info.lock().await;
+
+        // We have to freeze I/O (re-)submissions while doing that, to prevent
+        // an uncontrollable storm of I/O resubmissions in the case
+        // the persistent store is slow to response, or has failed.
+        self.set_nexus_io_mode(IoMode::Freeze).await;
```

Comment:
Should we freeze only when op is UpdateCond? Otherwise it seems not necessary?

Reply:
It won't hurt freeze for all conditions. It will prevent a device failure/retire while ETCD is being written.